packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -62,10 +62,30 @@ const variantsResolver = (props, styles, theme, name) => {
 const shouldForwardProp = (prop) =>
   prop !== 'styleProps' && prop !== 'theme' && prop !== 'isRtl' && prop !== 'sx' && prop !== 'as';
 
+const lowercaseFirstLetter = (string) => {
+  return string.charAt(0).toLowerCase() + string.slice(1);
+};
+
 const experimentalStyled = (tag, options, muiOptions = {}) => {
-  const name = muiOptions.muiName;
+  const componentName = muiOptions.name;
+  const componentSlot = muiOptions.slot;
   const skipSx = muiOptions.skipSx || false;
-  const defaultStyledResolver = styled(tag, { shouldForwardProp, label: name, ...options });
+
+  let displayName;
+  let name;
+  let className;
+
+  if (componentName && componentSlot) {
+    displayName = `${componentName}${componentSlot}`;
+    name = componentSlot === 'Root' ? `Mui${componentName}` : null;
+    className = `Mui${componentName}-${lowercaseFirstLetter(componentSlot)}`;
```

Comment:
```suggestion
    className = `Mui${componentName}-${lowercaseFirstLetter(componentSlot || 'Root')}`;
```

Reply:
I have been meaning to ask, what is the rationale behind `lowerCaseFirstLetter`ing the slot?